pkg/image/source.go

```diff
@@ -105,9 +113,21 @@ func detectSource(fs afero.Fs, userInput string) (Source, string, error) {
 			return UnknownSource, "", fmt.Errorf("unable to expand potential home dir expression: %w", err)
 		}
 	case UnknownSource:
-		if isDockerReference(userInput) {
-			// ignore any source hint since the source is ultimately unknown, see if this could be a docker image
-			return DockerDaemonSource, userInput, nil
+		// ignore any source hint since the source is ultimately unknown, see if this could be a docker image
+		if isRegistryReference(userInput) {
+			// verify that the docker daemon is accessible before assuming we can suggest to use it
+			dockerClient, err := docker.GetClient()
+			if err == nil {
+				ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
```

Comment:
I missed this, but why is the timeout 1 second? I can see how this could break and cause issues with no way to workaround it. In corporate environments with VPNs required to access remote servers I would expect this to take more than a second

Reply:
that's a good point, I'll bump this to, say, 10?

Reply:
I think bumping this makes sense, but I call these magical numbers because we are guessing and not allowing other values to be set. We can go with 10, but I think we need to revisit this with a separate issue to make it configurable.